lwip: version up 2.0.2

sinzah opened the pull request:
**# lwIP version up - initial updates from 1.4.1 to 2.0.2**

To support IPv6 completely, lwIP version should be updated to later than 2.0.2
So, this pull request is registered to version up TizenRT lwip stack from 1.4.1 to 2.0.2
This is a base pull request to go next step for merging IPv6 Ready patches

**Major Changes:**
1. netdb has been changed from libc to lwip
 - libc netdb library doesn't support IPv6 functionalities so removed
 - netdb functions are re-mapped to lwip netdb functions (see below example in bsd_socket_api.c)
`struct hostent *gethostbyname(const char *name)
{
     return lwip_gethostbyname(name);
}`
 - e.g., gethostbyname, getaddrinfo, freeaddrinfo and so on

>  (Note that)
   include header is not changed, user can use netdb.h without fixes in application
   but, CONFIG_LIBC_NETDB is not supported anymore, CONFIG_NET_LWIP_NETDB will replace libc option

2. lwIP Kconfig menu newly added for IPv6
 - ND6 (Neighbor Discovery for IPv6)
 - MLD (Mutlicast Listener Discovery for IPv6)
 - ICMPv6 (Internet Control Message Protocol for IPv6)

>  (Note that)
    IPv6 functionalities will completely support with another IPv6 Ready Pull request
    Currently, basic operation of IPv6 has no problem through SLAAC (Stateless Address Autoconfiguration) but not fully satisfied IPv6 Ready test

**Minor Changes:**
1. Add condition on le_tc testcase to provide enhanced granularity
 - AF_UNIX, AF_X25, AF_PACKET, AF_LOCAL domains are not supported so add condition to them

2. iotjs and iotivity build error fixes after version-up
 - Some IPv6 macro is used without checking configuration
 - un-supported netdb MACRO is used so add condition to resolve build error


**Test Results**

Basic regression test  (IPv4, configuration : artik053/nettest)
 - Wi-Fi : OK (join, leave, scan)
 - DHCP client : OK
 - DHCP server : OK
 - DNS client : OK
 - Socket Test : OK (through iperf application)
 - NTP client : OK
 - mDNS : OK (discover, resolve)

Build Sanity Test : OK
 - artik053/minimal
 - artik053/tc
 - artik053/st_things
 - artik053/iotjs
 - qemu/tc_64k

seinfra commented:
Code Rule Check OK.

seinfra commented:
Code Rule Check OK.

seinfra commented:
Code Rule Check OK.

pillip8282 commented:
@sinzah 
I found that gethostbyname() calls lwip_gethostbyname() directly which is defined os layer. In that case it can be a problem if a protected mode is enable. Did you consider it?

sinzah commented:
@pillip8282 

Thanks for comment

As you knew that the lwIP netdb API doesn't call TCP/IP thread API directly.
It sends a TCPIP_MSG_CALLBACK message to TCP/IP thread, and then wait sem signal from TCP/IP thread.
The call stack from gethostbyname to lwip_gethostbyname will be application stack, however
after sent TCPIP_MSG_CALLBACK message, TCP/IP stack will handle DNS procedure on its own stack.

I mean that lwIP netdb implementation considering the separation between application and lwIP TCP/IP thread.

Could you tell me your opinion and let me know if you have any kinds of question.

seinfra commented:
Code Rule Check OK.

pillip8282 commented:
@sinzah 

Current implementation will have a problem if protecteced mode(i.e syscall) is enabled. so netdb API should support syscall.

sinzah commented:
@pillip8282 

Could you add comment for further steps?

![image](https://user-images.githubusercontent.com/2297211/36416998-11d7413a-166e-11e8-997e-0b436afd3eab.png)

1. Could you share how to test protected build for ARTIK-053 or ARTIK-051?
 - Do I only need to check the API is working?
2. And which configuration should I use?
 - There is no BUILD_PROTECTED build enabled configuration
 - After I enabled it by manually, below build error occurs 
    compiler version : arm-none-eabi-gcc 4.9.3
    branch : master (commit-hash : cd0554dcb8)
CC:  proxies/PROXY__exit.c
/home/jseong/gitHub/TizenRT/os/include/unistd.h: In function '_exit':
proxies/PROXY__exit.c:10:1: error: 'noreturn' function does return [-Werror]
 }
 ^
cc1: all warnings being treated as errors
make[1]: *** [PROXY__exit.o] Error 1
make[1]: Leaving directory `/home/jseong/gitHub/TizenRT/os/syscall'
make: *** [syscall/libproxies.a] Error 2

Thanks

sunghan-chang commented on apps/system/utils/netcmd_ping.c:
```diff
@@ -193,8 +194,13 @@ nu_ping_recv(int s, struct timespec *ping_time)
+			printf("nu_ping_recv: error [%d]\n", errno);
+			break;
+		}
+		else if (len == 0) {
```
This is a coding rule violation. You should locate ```else if``` at same line of closing bracket.
```
} else if (len == 0) {
```

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/ethernet.h:
```diff
@@ -59,12 +59,27 @@
 
 #include <stdint.h>
 
+#if CONFIG_NET_LWIP
```
```#ifdef``` is correct.

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/api.h:
```diff
@@ -76,40 +79,85 @@ extern "C" {
 /* Helpers to process several netconn_types by the same code */
-#define NETCONNTYPE_GROUP(t)    (t&0xF0)
-#define NETCONNTYPE_DATAGRAM(t) (t&0xE0)
+#define NETCONNTYPE_GROUP(t)         ((t)&0xF0)
```
Please add spaces around ```&```.
```
((t) & 0xF0)
```

Same as line 113, 117, 118

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/api.h:
```diff
@@ -127,192 +201,217 @@ enum netconn_evt {
-/** A callback prototype to inform about events for a netconn */
-typedef void (*netconn_callback)(struct netconn *, enum netconn_evt, u16_t len);
+/* A callback prototype to inform about events for a netconn */
+typedef void (*netconn_callback) (struct netconn *, enum netconn_evt, u16_t len);
```
let's remove an added space

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/api.h:
```diff
@@ -127,192 +201,217 @@ enum netconn_evt {
 	struct socketlist *slist;
-	/** interoperability with tinyara network stack */
-	int crefs;			  /** for dup */
+	/* interoperability with tinyara network stack * */
```
Let's remove added *.

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/api.h:
```diff
@@ -127,192 +201,217 @@ enum netconn_evt {
-err_t netconn_delete(struct netconn *conn);
-/** Get the type of a netconn (as enum netconn_type). */
+struct netconn *netconn_new_with_proto_and_callback(enum netconn_type t, u8_t proto, netconn_callback callback);
+	err_t netconn_delete(struct netconn *conn);
```
please remove a tab at start of line

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/apps/snmp.h:
```diff
@@ -0,0 +1,143 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
```
2018

sinzah replied:
Fixed

sunghan-chang replied:
Could you check here again? Not fixed.

sunghan-chang commented on os/include/net/lwip/apps/snmp.h:
```diff
@@ -0,0 +1,143 @@
+	/** value ASN1 type */
+	u8_t type;
+	/** object value length */
+	u16_t value_len;
```
To use ```u8_t``` and ```u16_t```, you need to include some header.

sinzah replied:
@sunghan-chang 
lwip has arch.h headers for compatibility to various architecture.
please refer  os/include/net/lwip/arch.h

sunghan-chang commented on os/include/net/lwip/apps/snmp_core.h:
```diff
@@ -0,0 +1,364 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
```
2018. Please check all of new file.

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/apps/snmp_core.h:
```diff
@@ -0,0 +1,364 @@
+
+struct snmp_node_instance;
+
+typedef s16_t(*node_instance_get_value_method) (struct snmp_node_instance *, void *);
```
for typdef of fucntion prototype, ```return_type (*function_name)(argument);``` is right.
1. add a spave between return tye and function name
2. remove a space between function name and argument

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/debug.h:
```diff
@@ -78,50 +87,53 @@
+
+#ifndef LWIP_NOASSERT
+#define LWIP_ASSERT(message, assertion) do { if (!(assertion)) { \
+  LWIP_PLATFORM_ASSERT(message); }} while(0)
```
Let's use a tab for indentation. coding rule violation

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/debug.h:
```diff
@@ -78,50 +87,53 @@
+#error "If you want to use LWIP_DEBUG, LWIP_PLATFORM_DIAG(message) needs to be defined in your arch/cc.h"
+#endif
+#define LWIP_DEBUGF(debug, message) do { \
+                               if ( \
```
same as previous

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/def.h:
```diff
@@ -60,46 +66,23 @@ extern "C" {
 #define LWIP_MIN(x , y)  (((x) < (y)) ? (x) : (y))
 
+/* Get the number of entries in an array ('x' must NOT be a pointer!) */
+#define LWIP_ARRAYSIZE(x) (sizeof(x)/sizeof((x)[0]))
```
spaces around '/'

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/def.h:
```diff
@@ -60,46 +66,23 @@ extern "C" {
+#define LWIP_ARRAYSIZE(x) (sizeof(x)/sizeof((x)[0]))
+
+/** Create u32_t value from bytes */
+#define LWIP_MAKEU32(a,b,c,d) (((u32_t)((a) & 0xff) << 24) | \
```
space after ','

sinzah replied:
Fixed

sunghan-chang commented on os/include/net/lwip/def.h:
```diff
@@ -110,32 +93,63 @@ extern "C" {
-					 (((x) & 0xff0000UL) >> 8) | \
-					 (((x) & 0xff000000UL) >> 24))
+#define PP_HTONL(x) ((((x) & 0x000000ffUL) << 24) | \
+                     (((x) & 0x0000ff00UL) <<  8) | \
```
spaces at start of line are not allowed to coding rule.

sinzah replied:
Fixed

sunghan-chang commented:
@sinzah There are some coding rule violations. Could you check all of lwip source codes through our coding rule checker?

sinzah commented:
@sunghan-chang 

Thanks for reviews
Is there script for adapting TizenRT coding style?

sunghan-chang commented:
@sinzah There is the formatter.sh on os/tools. But be aware of abnormal operation for type casting.
Our coding rule says we should not add a space after type casting. But formatter makes it sometimes.
And something is missing.

seinfra commented:
Code Rule Check OK.

seinfra commented:
Code Rule Check Result:
os/include/net/lwip/debug.h:97: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/debug.h:97: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/debug.h:116: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
os/include/net/lwip/debug.h:116: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/debug.h:116: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/debug.h:130: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/debug.h:133: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/def.h:74: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/def.h:74: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/def.h:75: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/def.h:75: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/def.h:76: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/def.h:76: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/inet.h:84: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/inet.h:85: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/inet.h:86: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/inet.h:87: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/inet.h:101: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/inet.h:104: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/inet.h:158: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/inet.h:158: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/inet.h:159: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/inet.h:159: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/inet.h:160: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/inet.h:160: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/inet.h:160: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
os/include/net/lwip/inet.h:162: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/inet.h:162: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/inet.h:163: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/inet.h:163: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/inet.h:164: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/inet.h:164: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/inet.h:164: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
os/include/net/lwip/ip.h:84: ERROR: [SPC_M_OPR] space required after that ';' (ctx:WxV)
os/include/net/lwip/ip.h:326: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/ip.h:326: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip4_addr.h:86: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip4_addr.h:161: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/ip4_addr.h:169: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip4_addr.h:180: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/ip4_addr.h:180: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip4_addr.h:181: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/ip4_addr.h:181: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip4_addr.h:182: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/ip4_addr.h:182: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip6_addr.h:90: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip6_addr.h:113: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip6_addr.h:118: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip6_addr.h:121: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/include/net/lwip/ip6_addr.h:122: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/ip6_addr.h:122: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip6_addr.h:123: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/ip6_addr.h:123: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip6_addr.h:124: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/ip6_addr.h:124: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip6_addr.h:124: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
os/include/net/lwip/ip6_addr.h:124: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/ip6_addr.h:124: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip6_addr.h:132: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip6_addr.h:138: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip6_addr.h:213: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip6_addr.h:222: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip6_addr.h:224: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
os/include/net/lwip/ip6_addr.h:226: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/include/net/lwip/ip6_addr.h:231: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip6_addr.h:261: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip6_addr.h:262: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/ip6_addr.h:262: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip_addr.h:77: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip_addr.h:98: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip_addr.h:119: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/ip_addr.h:119: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/ip_addr.h:119: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/ip_addr.h:119: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/ip_addr.h:119: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/ip_addr.h:119: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/ip_addr.h:139: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:140: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:157: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:160: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:165: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/include/net/lwip/ip_addr.h:167: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:170: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:173: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:176: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:181: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/include/net/lwip/ip_addr.h:183: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:188: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:191: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:194: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:196: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/include/net/lwip/ip_addr.h:198: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:200: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/include/net/lwip/ip_addr.h:202: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:204: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/include/net/lwip/ip_addr.h:206: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:208: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/include/net/lwip/ip_addr.h:210: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:245: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:248: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:265: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/ip_addr.h:271: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/ip_addr.h:335: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/ip_addr.h:335: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/ip_addr.h:335: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/ip_addr.h:335: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:58: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:58: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:58: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:63: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:63: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:63: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:81: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:81: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:81: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:105: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:105: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:105: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/memp.h:107: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/memp.h:109: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/memp.h:111: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netbuf.h:118: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/netif.h:403: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/netif.h:484: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/netif.h:484: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/netif.h:491: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/netif.h:491: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/netif.h:497: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/netif.h:497: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/netif.h:499: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/netif.h:499: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/netif.h:524: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/netif.h:524: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/pbuf.h:224: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/pbuf.h:224: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/priv/tcp_priv.h:128: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/priv/tcp_priv.h:128: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/priv/tcp_priv.h:191: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/priv/tcp_priv.h:193: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/priv/tcp_priv.h:201: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/include/net/lwip/priv/tcp_priv.h:208: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/include/net/lwip/priv/tcp_priv.h:233: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/include/net/lwip/priv/tcp_priv.h:240: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/include/net/lwip/priv/tcp_priv.h:247: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/priv/tcp_priv.h:306: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/priv/tcp_priv.h:307: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/priv/tcp_priv.h:308: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/priv/tcp_priv.h:372: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/priv/tcp_priv.h:388: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/priv/tcp_priv.h:416: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/priv/tcpip_priv.h:80: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/priv/tcpip_priv.h:86: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/prot/dns.h:129: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/snmp.h:118: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/snmp.h:123: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/snmp.h:146: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/snmp.h:167: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/snmp.h:168: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/snmp.h:194: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/snmp.h:195: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/snmp.h:210: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/snmp.h:216: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sockets.h:202: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/sockets.h:428: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sockets.h:430: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sockets.h:430: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sockets.h:432: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sockets.h:432: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sockets.h:473: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/sockets.h:473: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/sockets.h:478: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/stats.h:327: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/sys.h:69: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sys.h:82: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sys.h:83: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sys.h:84: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sys.h:85: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sys.h:92: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sys.h:92: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sys.h:92: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sys.h:92: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/sys.h:281: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/sys.h:434: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/sys.h:443: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/sys.h:452: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/sys.h:461: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/tcp.h:384: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

seinfra commented:
Code Rule Check OK.

seinfra commented:
Code Rule Check Result:
os/include/net/lwip/debug.h:116: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/include/net/lwip/icmp.h:106: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/icmp.h:106: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/icmp.h:107: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/icmp.h:107: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/icmp.h:109: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
os/include/net/lwip/icmp.h:109: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/include/net/lwip/icmp.h:109: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/icmp.h:109: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/icmp.h:111: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
os/include/net/lwip/icmp.h:111: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/include/net/lwip/icmp.h:111: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/include/net/lwip/icmp.h:111: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/inet.h:84: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/inet.h:85: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/inet.h:86: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/inet.h:87: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/init.h:102: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/init.h:102: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip6_addr.h:122: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/ip6_addr.h:122: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip6_addr.h:123: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/ip6_addr.h:123: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip6_addr.h:124: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/ip6_addr.h:124: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/ip6_addr.h:224: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
os/include/net/lwip/ip6_addr.h:226: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/include/net/lwip/ip_addr.h:181: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/include/net/lwip/lwipopts.h:127: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/lwipopts.h:129: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/lwipopts.h:539: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/lwipopts.h:543: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/lwipopts.h:1066: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/net/lwip/netif/ppp/chap-new.h:102: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:102: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:102: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:103: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:103: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:103: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:104: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:104: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:105: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:108: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:108: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:108: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:109: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:118: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:118: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:118: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:119: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:119: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:119: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:120: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:120: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:120: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:121: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:124: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:124: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:124: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:125: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:131: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:131: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:131: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:132: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:132: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:132: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:133: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:133: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:133: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:134: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:137: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/chap-new.h:137: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/include/net/lwip/netif/ppp/chap-new.h:137: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:138: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/mppe.h:94: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/mppe.h:112: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/mppe.h:116: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/mppe.h:146: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/ppp.h:497: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/netif/ppp/ppp.h:497: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/ppp.h:497: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/netif/ppp/ppp.h:569: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/netif/ppp/ppp.h:569: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/ppp.h:578: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/netif/ppp/ppp.h:578: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/ppp.h:587: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/netif/ppp/ppp.h:587: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/ppp.h:699: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/netif/ppp/ppp.h:699: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/ppp.h:703: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/netif/ppp/ppp.h:703: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/ppp_impl.h:542: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/netif/ppp/ppp_impl.h:543: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/include/net/lwip/netif/ppp/ppp_impl.h:555: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/ppp_impl.h:556: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/netif/ppp/ppp_impl.h:557: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/opt.h:121: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/opt.h:121: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/opt.h:121: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/opt.h:121: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/opt.h:129: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/opt.h:129: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/opt.h:129: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/opt.h:129: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/opt.h:1255: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/net/lwip/priv/memp_priv.h:123: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/priv/memp_priv.h:123: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/priv/memp_priv.h:123: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/priv/memp_priv.h:132: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/priv/memp_priv.h:132: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/priv/memp_priv.h:132: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/net/lwip/priv/memp_std.h:78: ERROR: [SPC_M_OPR] spaces required around that '==' (ctx:VxV)
os/include/net/lwip/prot/dhcp.h:110: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/prot/etharp.h:85: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/prot/ethernet.h:77: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/prot/ethernet.h:84: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/prot/ethernet.h:94: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/prot/ethernet.h:102: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/prot/ethernet.h:110: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/prot/icmp6.h:177: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/prot/nd6.h:115: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/net/lwip/udp.h:158: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/net/lwip/src/api/api_msg.c:101: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/net/lwip/src/api/netdb.c:497: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/api/netdb.c:497: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/api/sockets.c:131: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/net/lwip/src/api/sockets.c:158: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/api/sockets.c:158: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/api/sockets.c:160: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/api/sockets.c:160: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/api/sockets.c:167: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/api/sockets.c:167: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/api/sockets.c:169: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/api/sockets.c:169: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/dns.c:188: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/net/lwip/src/core/init.c:143: ERROR: [SPC_M_OPR] spaces required around that '<=' (ctx:VxV)
os/net/lwip/src/core/init.c:190: ERROR: [SPC_M_OPR] spaces required around that '==' (ctx:VxV)
os/net/lwip/src/core/init.c:193: ERROR: [SPC_M_OPR] spaces required around that '==' (ctx:VxV)
os/net/lwip/src/core/ipv4/autoip.c:100: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
os/net/lwip/src/core/ipv4/autoip.c:101: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/autoip.c:101: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv4/autoip.c:102: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/autoip.c:102: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv4/autoip.c:103: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/autoip.c:103: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv4/autoip.c:104: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/autoip.c:104: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv4/autoip.c:104: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)
os/net/lwip/src/core/ipv4/autoip.c:113: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv4/autoip.c:114: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/autoip.c:114: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv4/etharp.c:147: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/etharp.c:147: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv4/igmp.c:720: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/igmp.c:721: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/ip4.c:110: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/ip4.c:110: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv4/ip4.c:722: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/ip4.c:723: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/ip4.c:938: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/ip4.c:939: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/ip4.c:970: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/ip4.c:971: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv4/ip4_frag.c:120: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv4/ip4_frag.c:121: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv4/ip4_frag.c:122: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/ipv6/ip6.c:755: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv6/ip6.c:756: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv6/ip6.c:869: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv6/ip6.c:870: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv6/ip6.c:914: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/ipv6/ip6.c:915: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/net/lwip/src/core/memp.c:86: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/lwip/src/core/memp.c:86: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/lwip/src/core/memp.c:86: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/lwip/src/core/memp.c:86: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/lwip/src/core/memp.c:86: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/lwip/src/core/memp.c:86: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/lwip/src/core/memp.c:90: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/lwip/src/core/memp.c:90: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/lwip/src/core/memp.c:90: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/lwip/src/core/snmp/snmp_asn1.c:62: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_asn1.c:63: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_asn1.c:64: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_core.c:199: ERROR: [SPC_M_OPR] spaces required around that '<=' (ctx:VxV)
os/net/lwip/src/core/snmp/snmp_mib2_icmp.c:67: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_mib2_interfaces.c:69: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_mib2_ip.c:69: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_mib2_system.c:68: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_mib2_tcp.c:70: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_mib2_udp.c:69: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:582: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:583: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:584: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:585: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:586: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:589: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:590: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:591: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:592: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:593: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:596: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:597: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:598: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/snmp/snmp_msg.c:599: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/net/lwip/src/core/tcp_in.c:519: ERROR: [BRC_M_SMT] else should follow close brace '}'
os/net/lwip/src/netif/ethernet.c:265: ERROR: [BRC_M_FTN] open brace '{' following function declarations go on the next line
os/net/lwip/src/netif/ethernetif.c:109: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:WxO)
os/net/lwip/src/netif/ethernetif.c:110: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:WxO)
os/net/lwip/src/netif/ethernetif.c:206: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:WxO)
os/net/socket/bsd_socket_api.c:248: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/socket/bsd_socket_api.c:248: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/socket/bsd_socket_api.c:253: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/socket/bsd_socket_api.c:253: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/socket/bsd_socket_api.c:263: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/net/socket/bsd_socket_api.c:268: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)

seinfra commented:
Code Rule Check Result:
os/include/net/lwip/netif/ppp/chap-new.h:102: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/include/net/lwip/netif/ppp/chap-new.h:104: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)

seinfra commented:
Code Rule Check OK.

seinfra commented:
Code Rule Check OK.

sinzah commented:
@sunghan-chang 

I've rebased my github to rearrage the commit, so some commits are disappeared.
Sorry for your confuse.

As results of code rule checker, hopefully every code rule is OK in my pull request.

If you have any kinds of concerns, please comment

sunghan-chang commented on os/include/net/ethernet.h:
```diff
@@ -1,6 +1,6 @@
 /****************************************************************************
  *
- * Copyright 2016 Samsung Electronics All Rights Reserved.
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
```
2016 is correct

sunghan-chang commented on os/include/net/if.h:
```diff
@@ -1,6 +1,6 @@
 /****************************************************************************
  *
- * Copyright 2016 Samsung Electronics All Rights Reserved.
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
```
revert

sunghan-chang commented on os/include/net/lwip/api.h:
```diff
@@ -1,6 +1,6 @@
 /****************************************************************************
  *
- * Copyright 2016 Samsung Electronics All Rights Reserved.
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
```
revert

sunghan-chang commented on os/include/net/lwip/def.h:
```diff
@@ -1,6 +1,6 @@
 /****************************************************************************
  *
- * Copyright 2016 Samsung Electronics All Rights Reserved.
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
```
Please check all of modified file. When you add new file, 2018 is correct. But if you modify existed file, previous year is correct. Because that shows when it was made (added).

sunghan-chang commented:
Above comment is a last to me. This PR is too huge so that it causes difficult to review.

sinzah commented:
@sunghan-chang 

After discussion of lwIP Maintainer, I will progress this PR again.

JeonginKim commented:
Network TC Failed. Please check it.

sunghan-chang commented:
Could you remove below commit? We already fixed it and can not allow to merge below commit .
```
mm: remove heapinfo_internal.h to prevent build breaks
```

sinzah commented:
@pillip8282 
I fixed conflicts and removed heapinfo_internal.h related commit.
If you merge on your branch. I will progress IPv6 ready pull request

sunghan-chang commented:
@sinzah Could you check my comment again? I again find year of copyright is 2016 in newly added file, like snmp.h.

sinzah commented:
@sunghan-chang 

I changed year of license from 2016 to 2018 for below newly added files (130 files).
os/include/net/lwip/apps/snmp.h
os/include/net/lwip/apps/snmp_core.h
os/include/net/lwip/apps/snmp_mib2.h
os/include/net/lwip/apps/snmp_opts.h
os/include/net/lwip/apps/snmp_scalar.h
os/include/net/lwip/apps/snmp_table.h
os/include/net/lwip/apps/snmp_threadsync.h
os/include/net/lwip/apps/snmpv3.h
os/include/net/lwip/autoip.h
os/include/net/lwip/dhcp6.h
os/include/net/lwip/dns.h
os/include/net/lwip/etharp.h
os/include/net/lwip/ethip6.h
os/include/net/lwip/icmp.h
os/include/net/lwip/icmp6.h
os/include/net/lwip/igmp.h
os/include/net/lwip/inet.h
os/include/net/lwip/inet_chksum.h
os/include/net/lwip/ip.h
os/include/net/lwip/ip4.h
os/include/net/lwip/ip4_addr.h
os/include/net/lwip/ip4_frag.h
os/include/net/lwip/ip6.h
os/include/net/lwip/ip6_addr.h
os/include/net/lwip/ip6_frag.h
os/include/net/lwip/ip_addr.h
os/include/net/lwip/mld6.h
os/include/net/lwip/nd6.h
os/include/net/lwip/netdb.h
os/include/net/lwip/netif/ethernet.h
os/include/net/lwip/netif/ppp/ccp.h
os/include/net/lwip/netif/ppp/chap-md5.h
os/include/net/lwip/netif/ppp/chap-new.h
os/include/net/lwip/netif/ppp/chap_ms.h
os/include/net/lwip/netif/ppp/eap.h
os/include/net/lwip/netif/ppp/ecp.h
os/include/net/lwip/netif/ppp/eui64.h
os/include/net/lwip/netif/ppp/fsm.h
os/include/net/lwip/netif/ppp/ipcp.h
os/include/net/lwip/netif/ppp/ipv6cp.h
os/include/net/lwip/netif/ppp/lcp.h
os/include/net/lwip/netif/ppp/magic.h
os/include/net/lwip/netif/ppp/mppe.h
os/include/net/lwip/netif/ppp/polarssl/arc4.h
os/include/net/lwip/netif/ppp/polarssl/des.h
os/include/net/lwip/netif/ppp/polarssl/md4.h
os/include/net/lwip/netif/ppp/polarssl/md5.h
os/include/net/lwip/netif/ppp/polarssl/sha1.h
os/include/net/lwip/netif/ppp/ppp.h
os/include/net/lwip/netif/ppp/ppp_impl.h
os/include/net/lwip/netif/ppp/ppp_opts.h
os/include/net/lwip/netif/ppp/pppapi.h
os/include/net/lwip/netif/ppp/pppcrypt.h
os/include/net/lwip/netif/ppp/pppdebug.h
os/include/net/lwip/netif/ppp/pppoe.h
os/include/net/lwip/netif/ppp/pppol2tp.h
os/include/net/lwip/netif/ppp/pppos.h
os/include/net/lwip/netif/ppp/upap.h
os/include/net/lwip/netif/ppp/vj.h
os/include/net/lwip/priv/api_msg.h
os/include/net/lwip/priv/memp_priv.h
os/include/net/lwip/priv/memp_std.h
os/include/net/lwip/priv/nd6_priv.h
os/include/net/lwip/priv/tcp_priv.h
os/include/net/lwip/priv/tcpip_priv.h
os/include/net/lwip/prot/autoip.h
os/include/net/lwip/prot/dhcp.h
os/include/net/lwip/prot/dns.h
os/include/net/lwip/prot/etharp.h
os/include/net/lwip/prot/ethernet.h
os/include/net/lwip/prot/icmp.h
os/include/net/lwip/prot/icmp6.h
os/include/net/lwip/prot/igmp.h
os/include/net/lwip/prot/ip.h
os/include/net/lwip/prot/ip4.h
os/include/net/lwip/prot/ip6.h
os/include/net/lwip/prot/mld6.h
os/include/net/lwip/prot/nd6.h
os/include/net/lwip/prot/tcp.h
os/include/net/lwip/prot/udp.h
os/include/net/lwip/timeouts.h
os/net/lwip/src/api/netdb.c
os/net/lwip/src/core/dns.c
os/net/lwip/src/core/inet_chksum.c
os/net/lwip/src/core/ip.c
os/net/lwip/src/core/ipv4/dhcp.c
os/net/lwip/src/core/ipv4/etharp.c
os/net/lwip/src/core/ipv4/ip4.c
os/net/lwip/src/core/ipv4/ip4_addr.c
os/net/lwip/src/core/ipv4/ip4_frag.c
os/net/lwip/src/core/ipv6/dhcp6.c
os/net/lwip/src/core/ipv6/ethip6.c
os/net/lwip/src/core/ipv6/ip6_frag.c
os/net/lwip/src/core/ipv6/mld6.c
os/net/lwip/src/core/ipv6/nd6.c
os/net/lwip/src/core/snmp/snmp_asn1.c
os/net/lwip/src/core/snmp/snmp_asn1.h
os/net/lwip/src/core/snmp/snmp_core.c
os/net/lwip/src/core/snmp/snmp_core_priv.h
os/net/lwip/src/core/snmp/snmp_mib2.c
os/net/lwip/src/core/snmp/snmp_mib2_icmp.c
os/net/lwip/src/core/snmp/snmp_mib2_interfaces.c
os/net/lwip/src/core/snmp/snmp_mib2_ip.c
os/net/lwip/src/core/snmp/snmp_mib2_snmp.c
os/net/lwip/src/core/snmp/snmp_mib2_system.c
os/net/lwip/src/core/snmp/snmp_mib2_tcp.c
os/net/lwip/src/core/snmp/snmp_mib2_udp.c
os/net/lwip/src/core/snmp/snmp_msg.c
os/net/lwip/src/core/snmp/snmp_msg.h
os/net/lwip/src/core/snmp/snmp_netconn.c
os/net/lwip/src/core/snmp/snmp_pbuf_stream.c
os/net/lwip/src/core/snmp/snmp_pbuf_stream.h
os/net/lwip/src/core/snmp/snmp_raw.c
os/net/lwip/src/core/snmp/snmp_scalar.c
os/net/lwip/src/core/snmp/snmp_table.c
os/net/lwip/src/core/snmp/snmp_threadsync.c
os/net/lwip/src/core/snmp/snmp_traps.c
os/net/lwip/src/core/snmp/snmpv3.c
os/net/lwip/src/core/snmp/snmpv3_dummy.c
os/net/lwip/src/core/snmp/snmpv3_mbedtls.c
os/net/lwip/src/core/snmp/snmpv3_priv.h
os/net/lwip/src/core/snmp2backup/Make.defs
os/net/lwip/src/core/snmp2backup/asn1_dec.c
os/net/lwip/src/core/snmp2backup/asn1_enc.c
os/net/lwip/src/core/snmp2backup/mib2.c
os/net/lwip/src/core/snmp2backup/mib_structs.c
os/net/lwip/src/core/snmp2backup/msg_in.c
os/net/lwip/src/core/snmp2backup/msg_out.c
os/net/lwip/src/core/timeouts.c
os/net/lwip/src/netif/ethernet.c

seinfra commented:
Target : [5d05e2d8ab26d90897104a057f8babd864a8fac3] - Code Rule Check OK.
